docker.go

```diff
@@ -961,11 +961,16 @@ func (p *DockerProvider) CreateContainer(ctx context.Context, req ContainerReque
 
 	sessionID := sessionID()
 
+	reaperOpts := containerOptions{}
+	for _, opt := range req.ReaperOptions {
+		opt(&reaperOpts)
+	}
+
 	var termSignal chan bool
 	// the reaper does not need to start a reaper for itself
-	isReaperContainer := strings.EqualFold(req.Image, reaperImage(req.ReaperImage))
+	isReaperContainer := strings.EqualFold(req.Image, reaperImage(reaperOpts.ImageName))
```

Comment:
This should satisfy what https://github.com/testcontainers/testcontainers-go/pull/720 is attempting

Reply:
Thanks for checking!